core/state/statedb.go

```diff
@@ -592,6 +597,11 @@ func (db *DB) GetRefund() uint64 {
 // Finalise finalises the state by removing the db destructed objects
 // and clears the journal as well as the refunds.
 func (db *DB) Finalise(deleteEmptyObjects bool) {
+	// Commit validator changes in cache to stateObjects
+	for addr, val := range db.stateValidators {
```

Comment:
This should be atomic right

Reply:
This is not committing to DB. It's only committing to the in-memory cache of stateObject, which will only be committed to DB when state.Commit() is called. So stateValidators is only a layer of cache for validatorWrapper object so it doesn't have to rlp to stateObject.code. stateObjects cache will be eventually committed to db when state.Commit() is called.